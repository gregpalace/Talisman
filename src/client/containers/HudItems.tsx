import React, { useState } from 'react';
import { connect } from 'react-redux';

import HealthBar from '../components/hud/healthBar';
import MagicBar from '../components/hud/magicBar';
import Inventory from '../components/hud/Inventory';

import '../../stylesheets/HUD-Styles/hudItems.scss';
import { IState } from '../../typings/PlayerTypes';

const mapStateToProps = (state: IState) => ({
  hp: state.player.hp,
  mp: state.player.mp,
  percentage: state.player.hp,
  inventorySlots: state.player.inventorySlots,
});

const HudItems = ({ hp, mp, inventorySlots, percentage }: any) => {
  const [inventoryOpen, setInventoryOpen] = useState(false);
  const showInventory = () => {
    setInventoryOpen(!inventoryOpen);
  };
  return (
    <div id='hud-items'>
      <HealthBar hp={hp} percentage={percentage} />
      <MagicBar mp={mp} percentage={percentage} />
      {inventoryOpen ? (
        <Inventory hp={hp} mp={mp} inventorySlots={inventorySlots} />
      ) : null}
      <button id='inventory-button' onClick={showInventory}>
        Show Inventory
      </button>
    </div>
  );
};

export default connect(mapStateToProps)(HudItems);
