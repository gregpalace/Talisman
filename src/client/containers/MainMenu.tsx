import React, { useState } from 'react';

import Flame from '../components/Flame';

import '../../stylesheets/mainMenu.scss';

const MainMenu = () => {
  const [isHovering, setHovering] = useState(false);

  const handleMouseHover = () => {
    return setHovering(!isHovering);
  };

  return (
    <div className='main-menu'>
      <header className='heading'>
        <img id='game-logo' src='../../assets/logo/logo.png' alt='Talisman' />
        <h1>Talisman</h1>
      </header>
      <br />
      <main className='menu-options'>
        <ul>
          <li>
            <a
              href='/game'
              onMouseEnter={handleMouseHover}
              onMouseLeave={handleMouseHover}
            >
              {' '}
              Start Game{' '}
              {isHovering && (
                <span>
                  <Flame />
                </span>
              )}
            </a>
          </li>
          <li>
            <a href='/game'>Continue Game</a>
          </li>
        </ul>
      </main>
    </div>
  );
};

export default MainMenu;
