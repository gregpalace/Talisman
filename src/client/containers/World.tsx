import React from 'react';
import { connect } from 'react-redux';
// import * as actions from '../actions/actions';

import Map from '../components/Map';
import Player from '../components/Player';
import { tiles } from '../../data/maps/1/map1';
import store from '../../store';

interface IPlayerState {
  readonly hp: number;
  readonly mp: number;
  readonly position: number[];
  readonly inventorySlots: number;
}

const mapStateToProps = (state: IPlayerState) => ({
  position: state.position,
});

const World = (props: any) => {
  store.dispatch({
    type: 'ADD_TILES',
    payload: {
      tiles,
    },
  });
  return (
    <div
      style={{
        position: 'relative',
        width: '1000px',
        height: '500px',
        margin: '20px auto',
      }}
    >
      <Map />
      <Player />
    </div>
  );
};

export default connect(mapStateToProps)(World);
