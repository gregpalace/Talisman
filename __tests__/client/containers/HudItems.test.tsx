import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HealthBar from '../../../src/client/components/hud/healthBar';
import MagicBar from '../../../src/client/components/hud/magicBar';
import { HudItems } from '../../../src/client/containers/HudItems';

describe('React unit tests', () => {
  let wrapper: any;
  let showInventory: any;

  const props = {
    hp: 100,
    mp: 100,
    percentage: 10,
  };

  beforeEach(() => {
    showInventory = jest.fn();
    wrapper = shallow(<HudItems {...props} showInventory={showInventory} />);
  });

  describe('<HudItems />', () => {
    xit('should render correclty', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render a health and magic bar', () => {
      expect(wrapper.find(HealthBar)).toBeDefined();
      expect(wrapper.find(MagicBar)).toBeDefined();
    });
    it('should render a healthbar that displays a players health', () => {
      expect(wrapper.find(HealthBar).prop('hp')).toBe(100);
      expect(wrapper.find(HealthBar).prop('hp')).not.toBe(0);
    });
    it('should render a magic bar that displays a players magic', () => {
      expect(wrapper.find(MagicBar).prop('mp')).toBe(100);
      expect(wrapper.find(MagicBar).prop('mp')).not.toBe(0);
    });
    it('should render a show inventory button', () => {
      expect(wrapper.find('button').prop('id')).toBe('inventory-button');
      expect(wrapper.find('button').prop('onClick')).toBeInstanceOf(Function);
    });
    it('should open a modal when the show inventory button is clicked', () => {
      wrapper.find('button').at(0).simulate('click');
      expect(showInventory).toHaveBeenCalled();
    });
  });
});
