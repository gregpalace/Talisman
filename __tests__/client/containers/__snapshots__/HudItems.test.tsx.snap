// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <HudItems /> should render correclty 1`] = `undefined`;
